tests/unit/filters/test_tilt.py

```diff
@@ -0,0 +1,282 @@
+    -------
+    @returns
+        A 3D numpy array depicting the two-sphere system, where air is 0, and object is 1.
+    """
+    # get the rotation object
+    rot_axis = rot_axis / np.linalg.norm(rot_axis)
+    rotation = R.from_rotvec(-omega * rot_axis)
+    # calculate the rotated sphere centers
+    # sphere 1
+    s1_rel = rotation.apply(s1_center_rel)
+    # sphere 2
+    s2_rel = rotation.apply(s2_center_rel)
+    # get the index grid
+    # NOTE: extend the range to make sure the sphere is not rotated out of the volume
+    # grid_x, grid_y, grid_z = np.mgrid[0:size, 0:size, 0:size]
+    # remapping to compensate for the strange coordinate system in tomopy projector
+    grid_y, grid_z, grid_x = np.mgrid[0:size, 0:size, 0:size]
+    # rescale to [-0.5, 0.5]
+    grid_x = grid_x / (size - 1) - 0.5
+    grid_y = -(grid_y / (size - 1) - 0.5)
+    grid_z = grid_z / (size - 1) - 0.5
+    # init volume
+    vol = np.zeros_like(grid_x)
+    # mark the voxels of sphere 1 to be 1
+    s1_dist_squared = (grid_x - s1_rel[0]) ** 2 + (grid_y - s1_rel[1]) ** 2 + (grid_z - s1_rel[2]) ** 2
+    r1_squared = s1_radius_rel**2
+    vol[s1_dist_squared < r1_squared] = 1.0
+    # mark the voxels of sphere 2 to be 2
+    s2_dist_squared = (grid_x - s2_rel[0]) ** 2 + (grid_y - s2_rel[1]) ** 2 + (grid_z - s2_rel[2]) ** 2
+    r2_squared = s2_radius_rel**2
+    vol[s2_dist_squared < r2_squared] = 1.0
+    return vol
+
+
+def virtual_cam(sample: np.ndarray, dynamic_range: int = 50_000) -> np.ndarray:
+    """
+    Projecting the 3D volume (sample) on to the x-z plane.
+
+    Parameter
+    ---------
+    @param sample:
+        3D numpy array representing the volume where air is 0 and object is 1
+    @param dynamic_range:
+        The dynamic range of the output image.
+
+    Returns
+    -------
+    @returns
+        2D numpy array representing the projected volume
+    """
+    img = sample.sum(axis=1)
+    img = (img.max() - img) / (img.max() - img.min())
+    return img * dynamic_range
+
+
+@pytest.mark.parametrize(
+    "tilt_reference",
+    [
+        0.0,
+        0.05,
+        0.1,
+        1.0,
+    ],
+)
+def test_calculate_tilt(tilt_reference):
+    """
+    Test the tilt correction with no tilt.
+    """
+    tilt_rad = np.radians(tilt_reference)
+    # get the rotation axis
+    # NOTE:
+    # we need to tilt the image -tilt in order to get the tilt angle as + value.
+    rot_aixs_tilted = get_tilted_rot_axis(tilt_inplane=-tilt_rad, tilt_outplane=0.0)
+    # radiograph at 0 deg
+    img0 = virtual_cam(two_sphere_system(0, rot_aixs_tilted, size=200))
+    # radiograph at 180 deg
+    img180 = virtual_cam(two_sphere_system(np.pi, rot_aixs_tilted, size=200))
+    # calculate the tilt angle
+    tilt_angle = calculate_tilt(img0, img180)
+    # verify
+    # NOTE: tolerance is set to half a pixel at the edge of the FOV
+    assert np.isclose(tilt_angle, tilt_reference, atol=np.degrees(0.5 / 100))
+
+
+def test_calculate_dissimilarity():
```

Comment:
For dissimilarity functions, I generally check that an image against itself is zero, and that two arbitrary images are abelian (order doesn't matter). Please add the latter, but I think you have to flip the sign on angle for it to work.

Reply:
the test has been added